.sandbox/provision/lifecycle-asg/on-create.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+ASG_NAME=test-windows-provision
+AVAILABILITY_ZONE=us-east-2a
+MAX_RETRIES=3
```

Comment:
Make these all configurable:

```sh
: ${ASG_NAME:=test-windows-provision}
: ${AVAILABILITY_ZONE:=us-east-2a}
: ${MAX_RETRIES:=3}
```

Please reconsider this, some configurations should not have default values, and as a reference solution, we should check if the required parameters are valid, if not, fail fast.